sklearn/manifold/tests/test_t_sne.py

```diff
@@ -317,10 +324,10 @@ def test_optimization_minimizes_kl_divergence():
 
 
 @pytest.mark.parametrize("method", ["exact", "barnes_hut"])
-def test_fit_csr_matrix(method):
+def test_fit_csr_matrix(method, global_dtype):
     # X can be a sparse matrix.
     rng = check_random_state(0)
-    X = rng.randn(50, 2)
+    X = rng.randn(50, 2).astype(global_dtype)
```

Comment:
Since the output of `truthworthiness` is a scalar, it would be nicer to use `pytest.approx` and an `assert`